internal/store/leaf/fsck.go

```diff
@@ -120,19 +181,23 @@ func (s *Store) fsckLoop(ctx context.Context, path string) error {
 
 		debug.Log("[%s] Checking %s", path, name)
 
-		msg, errs, err := s.fsckCheckEntry(ctx, name)
-		if errs == errsNil {
-			ctx = ctxutil.AddToCommitMessageBody(ctx, msg)
-		} else {
-			errcoll.WriteString(fmt.Errorf("failed to check %q:\n    %w\n", name, err).Error())
+		msg, err := s.fsckCheckEntry(ctx, name)
+		if err != nil {
+			warnings.WriteString(fmt.Errorf("failed to check %q:\n    %w\n", name, err).Error())
+
+			continue
 		}
+
+		ctx = ctxutil.AddToCommitMessageBody(ctx, msg)
 	}
 
-	if errcoll.Len() > 0 {
-		out.Errorf(ctx, errcoll.String())
+	// print out any deferred warnings (if any)
+	if warnings.Len() > 0 {
+		out.Errorf(ctx, warnings.String())
```

Comment:
Are these warnings or errors? 
I think we also have `out.Warningf` for the former.